kivy/graphics/instructions.pyx

```diff
@@ -456,7 +456,7 @@ cdef class Callback(Instruction):
         regarding that, please contact us.
 
     '''
-    def __init__(self, arg, **kwargs):
+    def __init__(self, arg=None, **kwargs):
```

Comment:
The fact that arg is now optional means that later in `apply` we need to check if it's not None and only execute the function if not None (`if self.func is not None and self.func(self):`).

Reply:
yep and if `self.func` is None `flag_update_done()` still need to be called, doesn't it?  I mean
 
```python
        if self.func is None or self.func(self):
            self.flag_update_done()
```

Reply:
and if `self.func` is None doesn't need to reset context. Am I correct?

Reply:
I don't think so. Because currently, if the `func` returns Falsey, then it indicates that no graphics was updated, hence why it diesn't call update. Similarly, if there's no func in the first place, then the we should assume we don't need to update.

Regarding the context reset, it still occurs even if the `func` returns Falsey so I'm asuming it should still happen. But I have not used `Callback` myself so I'm not really familiar with its usage.

Also, given that `arg` is now a optional keyword arg, we should rename it to `callback` because it wouldn't make sense to call `Callback(arg=...)`. But @tshirtman can chime in with what he thinks about these things.

Reply:
sorry I forgot about unittest.
I'm going to make it now

Reply:
Hmm, ok my bad. I misread `self.flag_update_done()` as `self.flag_update()` so you're right, without a callback, the update probably should be immediately considered done. @tshirtman is more familiar with the graphics internals so he can confirm.

Reply:
It's ok :)